SPEC.md

```diff
@@ -2601,6 +2602,20 @@ Integer ylen = length(ys) # 3
 Integer zlen = length(zs) # 0
 ```
 
+## Array[X] flatten(Array[Array[X]])
+
+Given an array of arrays, the `flatten` function concatentes all the
+member arrays in the order to appearance to give the result. For example:
```

Comment:
As written here, I think you'd need to call it twice to flatten an `Array[Array[Array[Int]]]`, but it might be nice for it to totally flatten any number of array nestings?

Reply:
Agree w/ @cjllanwarne - my assumption would be that `flatten` would return a completely flattened array

Reply:
This is certainly possible, although, that was not my original intention. I wanted to have a function like [flatten](http://www.scala-lang.org/api/2.12.3/scala/collection/immutable/List.html#flatten[B]:List[B]) in the scala standard library, or ocaml's [List.flatten](https://caml.inria.fr/pub/docs/manual-ocaml/libref/List.html). If we completely flatten all nesting levels, we would need another function for flattening only one nesting level, no?

Reply:
Actually my assumption was based on a bad memory of scala, I was thinking it fully flattened. You can see how much I have hands on keyboard these days :P

My main point stands either way though, I think it's worth being clear.